fish-rust/src/fd_monitor.rs

```diff
@@ -0,0 +1,573 @@
+use std::os::fd::{AsRawFd, RawFd};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+
+use self::fd_monitor::{c_void, new_fd_event_signaller, FdEventSignaller, ItemWakeReason};
+use crate::fd_readable_set::FdReadableSet;
+use crate::fds::AutoCloseFd2;
+use crate::ffi::void_ptr;
+use crate::flog::FLOG;
+use crate::wutil::perror;
+use cxx::UniquePtr;
+
+#[cxx::bridge]
+mod fd_monitor {
+    /// Reason for waking an item
+    #[repr(u8)]
+    #[cxx_name = "item_wake_reason_t"]
```

Comment:
interesting, this seems to do mostly the same things as a type alias on the C++ side.
Sadly it doesn't work for enum variants.

Reply:
Yeah. I tried defining them as consts i ln the shared cpp bridge section but constant aren’t supported.